File: 10550 Combination Lock/solution.c
#include <stdio.h>

int main() {
    int pos, num1, num2, num3;
    while(scanf("%d %d %d %d", &pos, &num1, &num2, &num3) && (pos != 0 || num1 != 0 || num2 != 0 || num3 != 0)) {
        int degree = 1080 + 9 * ((pos - num1 + 40) % 40 + (num2 - num1 + 40) % 40 + (num2 - num3 + 40) % 40);
        printf("%d\n", degree);
    }
    return 0;
}

File: 10550 Combination Lock/solution.java
import java.io.*;

public class Main {
	public static void main (String[] args) throws IOException {
		StreamTokenizer st = new StreamTokenizer(System.in);
		StringBuilder output = new StringBuilder();
		while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
		    int pos = (int)st.nval;
		    st.nextToken();
		    int num1 = (int)st.nval;
		    st.nextToken();
		    int num2 = (int)st.nval;
		    st.nextToken();
		    int num3 = (int)st.nval;
		    if(pos == 0 && num1 == 0 && num2 == 0 && num3 == 0)
		        break;
		    output.append(1080 + 9 * ((pos - num1 + 40) % 40 + (num2 - num1 + 40) % 40 + (num2 - num3 + 40) % 40) + "\n");
		}
		System.out.print(output);
	}
}
